gobblin-utility/src/main/java/gobblin/util/ParallelRunner.java

```diff
@@ -78,14 +80,35 @@
   private final ExecutorService executor;
   private final FileSystem fs;
 
-  private final List<Future<?>> futures = Lists.newArrayList();
+  private final List<NamedFuture> futures = Lists.newArrayList();
 
   private final Striped<Lock> locks = Striped.lazyWeakLock(Integer.MAX_VALUE);
 
+  private final FailPolicy failPolicy;
+
   public ParallelRunner(int threads, FileSystem fs) {
+    this(threads, fs, FailPolicy.FAIL_ONE_FAIL_ALL);
+  }
+
+  public ParallelRunner(int threads, FileSystem fs, FailPolicy failPolicy) {
     this.executor = Executors.newFixedThreadPool(threads,
         ExecutorsUtils.newThreadFactory(Optional.of(LOGGER), Optional.of("ParallelRunner")));
     this.fs = fs;
+    this.failPolicy = failPolicy;
+  }
+
+  public static enum FailPolicy {
```

Comment:
Can you add javadocs around the intent of these.  For ParallelRunner I think it is really important to be crystal clear what the expectations are.